spiketoolkit/comparison/comparisontools.py

```diff
@@ -0,0 +1,398 @@
+    """
+    This count all counting score possible lieke:
+      * TP: true positive
+      * CL: classification error
+      * FN: False negative
+      * FP: False positive
+      * TOT: 
+      * TOT_ST1: 
+      * TOT_ST2: 
+
+    Parameters
+    ----------
+    sorting1: SortingExtractor instance
+        The ground truth sorting.
+    
+    sorting2: SortingExtractor instance
+        The tested sorting.
+    
+    delta_tp: int
+        
+    unit_map12: dict
+        Dict of matching from sorting1 to sorting2.
+
+    Output
+    ----------
+    
+    counts: dict 
+        A dict containing all coutning
+    
+    labels_st1: np.array of str
+        Contain label for units of sorting 1
+    
+    labels_st2: np.array of str
+        Contain label for units of sorting 2
+    """
+    
+    unit1_ids = sorting1.getUnitIds()
+    unit2_ids = sorting2.getUnitIds()
+    labels_st1 = dict()
+    labels_st2 = dict()
+    N1 = len(unit1_ids)
+    N2 = len(unit2_ids)
+
+    # copy spike trains for faster access from extractors with memmapped data
+    sts1 = []
+    for u in sorting1.getUnitIds():
+        sts1.append(sorting1.getUnitSpikeTrain(u))
+    sts2 = []
+    for u in sorting2.getUnitIds():
+        sts2.append(sorting2.getUnitSpikeTrain(u))
+    
+    # Evaluate
+    for i,u1 in enumerate(unit1_ids):
+        lab_st1 = np.array(['UNPAIRED'] * len(sts1[i]))
+        labels_st1[u1] = lab_st1
+    for i,u2 in enumerate(unit2_ids):
+        lab_st2 = np.array(['UNPAIRED'] * len(sts2[i]))
+        labels_st2[u2] = lab_st2
+
+    for u_i, u1 in enumerate(sorting1.getUnitIds()):
+        u2 = unit_map12[u1]
+        if u2 !=-1:
+            lab_st1 = labels_st1[u1]
+            lab_st2 = labels_st2[u2]
+            mapped_st = sorting2.getUnitSpikeTrain(u2)
+            # from gtst: TP, TPO, TPSO, FN, FNO, FNSO
+            for sp_i, n_sp in enumerate(sts1[u_i]):
+                matches = (np.abs(mapped_st.astype(int)-n_sp)<=delta_tp//2)
+                if np.sum(matches) > 0:
+                        lab_st1[sp_i] = 'TP'
+                        lab_st2[np.where(matches)[0][0]] = 'TP'
+        else:
+            # Sam garcia note : I guess this is a mistake
+            lab_st1 = np.array(['FN'] * len(sts1[u_i]))
+            # Sam note : I would add this line
+            #�labels_st1[u1] = lab_st1
+
+    # find CL-CLO-CLSO
+    for u_i, u1 in enumerate(sorting1.getUnitIds()):
+        lab_st1 = labels_st1[u1]
+        #~ st1 = sorting1.getUnitSpikeTrain(u1)
+        st1 = sts1[u_i]
+        for l_gt, lab in enumerate(lab_st1):
+            if lab == 'UNPAIRED':
+                for u_j, u2 in enumerate(sorting2.getUnitIds()):
+                    if u2 in unit_map12.values() and unit_map12[u1] != -1:
+                        lab_st2 = labels_st2[u2]
+                        st2 = sts2[u_j]
+
+                        n_up = st1[l_gt]
+                        id_sp = np.where((st2.astype(int)-n_sp)<=delta_tp//2)[0]
```

Comment:
can you change these lines to:
```
matches = (np.abs(mapped_st.astype(int)-n_sp)<=delta_tp//2) 
if np.sum(matches) > 0:
    lab_st1[l_gt] = 'CL_' + str(u1) + '_' + str(u2)
    lab_st2[np.where(matches)[0][0]] = 'CL_' + str(u2) + '_' + str(u1)
```

Reply:
I think I made this change at the wring place. No ?